File: ios/FoodTracking/FoodTracking/Models/FoodEntry.swift
// FoodEntry Model - SwiftData equivalent of Room entity

import Foundation
import SwiftData

enum FoodSource: String, Codable {
    case photo = "PHOTO"
    case description = "DESCRIPTION"
    case search = "SEARCH"
    case favorites = "FAVORITES"
    case manual = "MANUAL"
}

@Model
final class FoodEntry {
    var id: UUID
    var name: String
    var calories: Int
    var fats: Float
    var proteins: Float
    var carbs: Float
    var emoji: String
    var weightGrams: Int
    var date: Date
    var imageUri: String?
    var source: String  // FoodSource raw value
    var healthKitRecordId: String?
    var createdAt: Date
    
    init(
        id: UUID = UUID(),
        name: String,
        calories: Int,
        fats: Float = 0,
        proteins: Float = 0,
        carbs: Float = 0,
        emoji: String = "🍽️",
        weightGrams: Int = 100,
        date: Date = Date(),
        imageUri: String? = nil,
        source: FoodSource = .manual,
        healthKitRecordId: String? = nil
    ) {
        self.id = id
        self.name = name
        self.calories = calories
        self.fats = fats
        self.proteins = proteins
        self.carbs = carbs
        self.emoji = emoji
        self.weightGrams = weightGrams
        self.date = date
        self.imageUri = imageUri
        self.source = source.rawValue
        self.healthKitRecordId = healthKitRecordId
        self.createdAt = Date()
    }
    
    var foodSource: FoodSource {
        FoodSource(rawValue: source) ?? .manual
    }
}

// Extension for date grouping
extension FoodEntry {
    var dateOnly: Date {
        Calendar.current.startOfDay(for: date)
    }
}

File: app/src/main/java/com/example/foodtracking/data/model/FoodEntry.kt
package com.example.foodtracking.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate
import java.time.LocalDateTime

/**
 * Represents a food entry logged by the user.
 */
@Entity(tableName = "food_entries")
data class FoodEntry(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String,
    val calories: Int,
    val fats: Float,
    val proteins: Float,
    val carbs: Float,
    val emoji: String = "🍽️",
    val weightGrams: Int = 100, // Portion weight in grams
    val date: LocalDate,
    val timestamp: LocalDateTime = LocalDateTime.now(),
    val imageUri: String? = null,
    val source: FoodSource = FoodSource.MANUAL,
    val healthConnectRecordId: String? = null // Health Connect record ID for sync tracking
) {
    /**
     * Scale the nutrition values based on a new weight.
     */
    fun scaledTo(newWeightGrams: Int): FoodEntry {
        if (weightGrams == 0 || newWeightGrams == weightGrams) return this
        val ratio = newWeightGrams.toFloat() / weightGrams.toFloat()
        return copy(
            weightGrams = newWeightGrams,
            calories = (calories * ratio).toInt(),
            proteins = proteins * ratio,
            carbs = carbs * ratio,
            fats = fats * ratio
        )
    }
}

/**
 * Source of the food entry.
 */
enum class FoodSource {
    PHOTO,
    DESCRIPTION,
    DATABASE,
    MANUAL
}
